src/RA_AchievementOverlay.cpp

```diff
@@ -430,7 +434,8 @@ BOOL AchievementOverlay::Update(ControllerInput* pInput, float fDelta, BOOL bFul
                     if ((*pnSelectedItem) < nNumLBs)
                     {
                         AddPage(OP_LEADERBOARD_EXAMINE);
-                        g_LBExamine.Initialize(g_LeaderboardManager.GetLB((*pnSelectedItem)).ID());
+                        auto* pLeaderboardManager = ra::services::ServiceLocator::Get<ra::services::ILeaderboardManager>();
+                        g_LBExamine.Initialize(pLeaderboardManager->GetLB((*pnSelectedItem)).ID());
```

Comment:
I'm a bit concerned about the raw pointer usage leakes. I'm not too sure what's going on but if this is similar .NET Core's `ILifetimeScope `service locator you should probably initialize it via `unique_ptr `so it gets deleted when the scope ends since C++ doesn't have Garbage Collection unless what you are using is from .NET/CLI.

The main concern is that raw pointers in native C++ don't have a concept of ownership, so unless you like reference counts (cycles), `unique_ptr `is probably the better option.

Reply:
The actual memory is owned by a `unique_ptr` in the ServiceLocator: https://github.com/RetroAchievements/RAIntegration/pull/66/files#diff-35f5e039eac7edca429cc193271a21caR86

The pointer here is the pointer returned from `unique_ptr.get()`. It's not meant to be referenced counted, and should not be deleted when it goes out of scope. It might as well be a reference (`auto&`), if not for the fact that it could be `nullptr` if no implementation is provided.